eos-gpt2mbr

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash -e
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+
+get_last_char() {
+	ret=$1
+	while [ "${#ret}" != "1" ]; do
+		ret=${ret#?}
+	done
+	echo $ret
+}
+
+# arguments checking
+if [ $# -lt 1 ]; then
+	echo "Usage: $0 <EOS partition>"
+	exit 1
+fi
+
+root_part=$1
+root_partno=$(get_last_char ${root_part})
+root_disk=${root_part%?}
+
+swap_partno=$((root_partno + 1))
+swap_part=${root_disk}${swap_partno}
+
+# check if we passed the full device file path
+if [ "${root_part//dev/}" = "${root_part}" ]; then
+	echo "eos-gpt2mbr: Specify the full path for the root device file (i.e. /dev/sda3)"
+	exit 1
+fi
+
+# check if we passed the device file for the partition and not for the disk
+case ${root_partno} in
+	*[!0-9]* )
+		echo "eos-gpt2mbr: Specify the root partition (i.e. /dev/sda3)"
+		exit 1
+		;;
+esac
+
+# check if the root partition does exist (if it exists we assume $root_disk does exist too)
+if [ ! -e "${root_part}" ]; then
+  echo "eos-gpt2mbr: $root_part doesn't exist"
+  exit 1
+fi
+
+# udev might still be busy probing the disk, meaning that it will be in use.
+udevadm settle
+
+# Save boot.img to be restored after we write the new MBR
+# as described in http://pete.akeo.ie/2014/05/compiling-and-installing-grub2-for.html
+dd if=${root_disk} of=boot.img conv=fsync bs=446 count=1
```

Comment:
should do this to /tmp or similar in case the current working dir is not writable